Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/LiveConnections/AutoReconnectLiveConnection.cs

```diff
@@ -50,15 +50,18 @@ public async UniTask SendAsync(MemoryWrap data, CancellationToken token)
             }
         }
 
-        public async UniTask<MemoryWrap> ReceiveAsync(CancellationToken token)
+        public async UniTask<EnumResult<MemoryWrap, IArchipelagoLiveConnection.ReceiveResponse>> ReceiveAsync(CancellationToken token)
         {
-            try { return await origin.ReceiveAsync(token); }
-            catch (ConnectionClosedException)
+            var result = await origin.ReceiveAsync(token);
+
+            if (result.Error?.State is IArchipelagoLiveConnection.ReceiveResponse.ConnectionClosed)
             {
                 log("Connection error on receiving, ensure to reconnect...");
                 await EnsureReconnectAsync(token);
```

Comment:
`EnsureReconnectAsync` should not throw exceptions either to follow the scheme.
Check `WebSocketArchipelagoLiveConnection.ConnectAsync`